core/src/data/value/error.rs

```diff
@@ -95,6 +105,9 @@ pub enum ValueError {
     #[error("literal cast failed to date: {0}")]
     LiteralCastToDateFailed(String),
 
+    #[error("literal cast from {1:#?} to ({0:#?} failed")]
+    LiteralCastToDataTypeFailed(DataType, String),
```

Comment:
1. You can use
`#[error("literal cast from {1} to ({0} failed")]`
`{:#?}` does not use `Display` so it would return verbose result.

2. Applying this only for INT128 first
The decision to add `LiteralCastToDataTypeFailed` is really nice, thanks.
However, it's perfect to start with applying to int128 but it would be better to make a separate PR for applying this to existing data types (e.g. INT8)

Reply:
ok.  sounds good.  I will try to make sure we aren't using LIteralCastToDatatypeFailed for anything but Int128 at this time.